networkx/algorithms/cycles.py

```diff
@@ -463,3 +465,157 @@ def find_cycle(G, source=None, orientation='original'):
+            # orthogonal to the newly found cycle
+            set_orth = _update_orth(set_orth, comp, new_cycle, k)
+
+        cbtotal += cb
+
+    return cbtotal
+
+
+def _update_orth(set_orth, graph, new_cycle, k):
+    """
+    Updates the set of vectors orthogonal to the first k basis cycles,
+    as per [p. 336, 1]
+    """
+    for orth in set_orth[k + 1:]:
+        if len(set.intersection(orth, new_cycle)) % 2:
+            orth = orth.symmetric_difference(set_orth[k])
+        else:
+            pass
+
+    return set_orth
+
+
+def _min_cycle(G, orth, weight=None):
+    """
+    Computes the minimum weight cycle in G, orthogonal to the vector orth
+    as per [p. 338, 1]
+    """
+    T = nx.Graph()
+
+    nodes_idx = {node: idx for idx, node in enumerate(G.nodes())}
+    idx_nodes = {idx: node for node, idx in nodes_idx.items()}
+
+    nnodes = len(nodes_idx)
+
+    # Add 2 copies of each edge in G to T. If edge is in orth, add cross edge;
+    # otherwise in-plane edge
+    if weight is not None:
+        for u, v in G.edges():
+            uidx, vidx = nodes_idx[u], nodes_idx[v]
+            edge_w = G[u][v][weight]
+            if frozenset((u, v)) in orth:
+                T.add_edges_from(
+                    [(uidx, nnodes + vidx), (nnodes + uidx, vidx)], weight=edge_w)
+
+            else:
+                T.add_edges_from(
+                    [(uidx, vidx), (nnodes + uidx, nnodes + vidx)], weight=edge_w)
+
+        all_shortest_pathlens = nx.shortest_path_length(T, weight='weight')
+
+    else:
+        for u, v in G.edges():
+            uidx, vidx = nodes_idx[u], nodes_idx[v]
+            if frozenset((u, v)) in orth:
+                T.add_edges_from(
+                    [(uidx, nnodes + vidx), (nnodes + uidx, vidx)])
+
+            else:
+                T.add_edges_from(
+                    [(uidx, vidx), (nnodes + uidx, nnodes + vidx)])
+
+        all_shortest_pathlens = nx.shortest_path_length(T)
+
+    cross_paths_w_lens = {
+        n: all_shortest_pathlens[n][nnodes + n] for n in range(nnodes)}
+
+    # Now compute shortest paths in T, which translates to cyles in G
+    min_path_startpoint = min(cross_paths_w_lens, key=cross_paths_w_lens.get)
+    min_path = nx.shortest_path(
+        T, source=min_path_startpoint, target=nnodes + min_path_startpoint, weight='weight')
+
+    # Now we obtain the actual path, re-map nodes in T to those in G
+    min_path_nodes = [
+        node if node < nnodes else node - nnodes for node in min_path]
+    # Now remove the edges that occur two times
+    mcycle_pruned = _path_to_cycle(min_path_nodes)
+
+    return {frozenset((idx_nodes[u], idx_nodes[v])) for u, v in mcycle_pruned}
+
+
+def _path_to_cycle(path):
+    """
+    Removes the edges from path that occur even number of times.
+    Returns a set of edges
+    """
+    edges = []
+    start = path[0]
+
+    for elem in path[1:]:
+        edge = frozenset((start, elem))
+        if edge in edges:
```

Comment:
I still think there's a better way of doing this. How about something like

    edges_to_keep = defaultdict(lambda: False)
    for edge in pairwise(path):
        # Toggle whether to keep the current edge.
        edges_to_keep[edge] ^= True
    return {e for e, keep in edges_to_keep.items() if keep}